cmd/noms/noms_commit.go

```diff
@@ -32,11 +32,13 @@ func setupCommitFlags() *flag.FlagSet {
 	commitFlagSet := flag.NewFlagSet("commit", flag.ExitOnError)
 	commitFlagSet.BoolVar(&allowDupe, "allow-dupe", false, "creates a new commit, even if it would be identical (modulo metadata and parents) to the existing HEAD.")
 	spec.RegisterCommitMetaFlags(commitFlagSet)
+	spec.RegisterVerboseFlags(commitFlagSet)
 	return commitFlagSet
 }
 
 func runCommit(args []string) int {
-	ds, err := spec.GetDataset(args[len(args)-1])
+	resolver := spec.NewResolver()
```

Comment:
Let's walk thru this in person. Hiding resolution in the existing methods is problematic because there are many call points in the code where a config file should not interfere. 

Making the resolver stateless also raises some issues. I think I've got it down to a simple one-liner now, but would love to talk over all the issues with you do determine if there's a better solution.